internal/services/containers/kubernetes_cluster_validate.go

```diff
@@ -26,16 +26,10 @@ func validateKubernetesCluster(d *pluginsdk.ResourceData, cluster *managedcluste
 				dockerBridgeCidr := profile["docker_bridge_cidr"].(string)
 				dnsServiceIP := profile["dns_service_ip"].(string)
 				serviceCidr := profile["service_cidr"].(string)
-				podCidr := profile["pod_cidr"].(string)
 				podCidrs := profile["pod_cidrs"].([]interface{})
 				serviceCidrs := profile["service_cidrs"].([]interface{})
 				isServiceCidrSet := serviceCidr != "" || len(serviceCidrs) != 0
 
-				// Azure network plugin is not compatible with pod_cidr
-				if podCidr != "" && networkPlugin == "azure" {
-					return fmt.Errorf("`pod_cidr` and `azure` cannot be set together")
-				}
-
```

Comment:
Why has this been removed?

Reply:
From [this document](https://learn.microsoft.com/en-us/azure/aks/azure-cni-powered-by-cilium), the cli command used to create the cluster is

```bash
az aks create -n <clusterName> -g <resourceGroupName> -l <location> \
  --max-pods 250 \
  --node-count 2 \
  --network-plugin azure \
  --network-plugin-mode overlay \
  --pod-cidr 192.168.0.0/16 \
  --vnet-subnet-id /subscriptions/<subscriptionId>/resourceGroups/<resourceGroupName>/providers/Microsoft.Network/virtualNetworks/<vnetName>/subnets/nodesubnet \
  --enable-cilium-dataplane
``` 

It uses the pod cider and `azure` network plugin at the same time. And even no pod cidr is provided, the server will return a default pod cidr.

Reply:
I think more needs to be done here because I get a diff when I try to create a cluster with the following network_profile
```
  network_profile {
    pod_cidr            = "192.168.0.0/16"
    network_plugin      = "azure"
  }
```

```
=== CONT  TestAccKubernetesCluster_networkPluginMode
    testcase.go:110: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
        -/+ destroy and then create replacement
        
        Terraform will perform the following actions:
        
          # azurerm_kubernetes_cluster.test must be replaced
        -/+ resource "azurerm_kubernetes_cluster" "test" {
              - enable_pod_security_policy          = false -> null
              ~ fqdn                                = "acctestaks221206104426356817-a1c75092.hcp.westcentralus.azmk8s.io" -> (known after apply)
              + http_application_routing_zone_name  = (known after apply)
              ~ id                                  = "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/acctestRG-aks-221206104426356817/providers/Microsoft.ContainerService/managedClusters/acctestaks221206104426356817" -> (known after apply)
              ~ kube_admin_config                   = (sensitive value)
              + kube_admin_config_raw               = (sensitive value)
              ~ kube_config                         = (sensitive value)
              ~ kube_config_raw                     = (sensitive value)
              ~ kubernetes_version                  = "1.23.12" -> (known after apply)
              - local_account_disabled              = false -> null
                name                                = "acctestaks221206104426356817"
              ~ node_resource_group                 = "MC_acctestRG-aks-221206104426356817_acctestaks221206104426356817_westcentralus" -> (known after apply)
              - oidc_issuer_enabled                 = false -> null
              + oidc_issuer_url                     = (known after apply)
              ~ portal_fqdn                         = "acctestaks221206104426356817-a1c75092.portal.hcp.westcentralus.azmk8s.io" -> (known after apply)
              + private_dns_zone_id                 = (known after apply)
              + private_fqdn                        = (known after apply)
                # (10 unchanged attributes hidden)
        
              + auto_scaler_profile {
                  + balance_similar_node_groups      = (known after apply)
                  + empty_bulk_delete_max            = (known after apply)
                  + expander                         = (known after apply)
                  + max_graceful_termination_sec     = (known after apply)
                  + max_node_provisioning_time       = (known after apply)
                  + max_unready_nodes                = (known after apply)
                  + max_unready_percentage           = (known after apply)
                  + new_pod_scale_up_delay           = (known after apply)
                  + scale_down_delay_after_add       = (known after apply)
                  + scale_down_delay_after_delete    = (known after apply)
                  + scale_down_delay_after_failure   = (known after apply)
                  + scale_down_unneeded              = (known after apply)
                  + scale_down_unready               = (known after apply)
                  + scale_down_utilization_threshold = (known after apply)
                  + scan_interval                    = (known after apply)
                  + skip_nodes_with_local_storage    = (known after apply)
                  + skip_nodes_with_system_pods      = (known after apply)
                }
        
              ~ default_node_pool {
                  - enable_auto_scaling          = false -> null
                  - enable_host_encryption       = false -> null
                  - enable_node_public_ip        = false -> null
                  - fips_enabled                 = false -> null
                  ~ kubelet_disk_type            = "OS" -> (known after apply)
                  - max_count                    = 0 -> null
                  ~ max_pods                     = 30 -> (known after apply)
                  - min_count                    = 0 -> null
                    name                         = "default"
                  ~ node_labels                  = {} -> (known after apply)
                  - only_critical_addons_enabled = false -> null
                  ~ orchestrator_version         = "1.23.12" -> (known after apply)
                  ~ os_disk_size_gb              = 128 -> (known after apply)
                  ~ os_sku                       = "Ubuntu" -> (known after apply)
                  + workload_runtime             = (known after apply)
                    # (7 unchanged attributes hidden)
                }
        
              ~ identity {
                  ~ principal_id = "28e62034-c843-4e3a-9640-29cc147c1380" -> (known after apply)
                  ~ tenant_id    = "26e25406-6564-4a26-98ee-c71ba03235ad" -> (known after apply)
                    # (1 unchanged attribute hidden)
                }
        
              ~ kubelet_identity {
                  ~ client_id                 = "84950e58-f0ba-426f-a190-5dbde10f115f" -> (known after apply)
                  ~ object_id                 = "b25e62b0-010e-429e-9e30-9e21d5927d3b" -> (known after apply)
                  ~ user_assigned_identity_id = "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/MC_acctestRG-aks-221206104426356817_acctestaks221206104426356817_westcentralus/providers/Microsoft.ManagedIdentity/userAssignedIdentities/acctestaks221206104426356817-agentpool" -> (known after apply)
                }
        
              ~ network_profile {
                  ~ dns_service_ip     = "10.0.0.10" -> (known after apply)
                  ~ docker_bridge_cidr = "172.17.0.1/16" -> (known after apply)
                  ~ ip_versions        = [
                      - "IPv4",
                    ] -> (known after apply)
                  + network_mode       = (known after apply)
                  + network_policy     = (known after apply)
                  + pod_cidr           = "192.168.0.0/16" # forces replacement
                  ~ pod_cidrs          = [] -> (known after apply)
                  ~ service_cidr       = "10.0.0.0/16" -> (known after apply)
                  ~ service_cidrs      = [
                      - "10.0.0.0/16",
                    ] -> (known after apply)
                    # (3 unchanged attributes hidden)
        
                  ~ load_balancer_profile {
                      ~ effective_outbound_ips      = [
                          - "/subscriptions/1a6092a6-137e-4025-9a7c-ef77f76f2c02/resourceGroups/MC_acctestRG-aks-221206104426356817_acctestaks221206104426356817_westcentralus/providers/Microsoft.Network/publicIPAddresses/3c905616-d322-460b-8a07-1f81dea93883",
                        ] -> (known after apply)
                      ~ idle_timeout_in_minutes     = 0 -> (known after apply)
                      ~ managed_outbound_ip_count   = 1 -> (known after apply)
                      ~ managed_outbound_ipv6_count = 0 -> (known after apply)
                      ~ outbound_ip_address_ids     = [] -> (known after apply)
                      ~ outbound_ip_prefix_ids      = [] -> (known after apply)
                      ~ outbound_ports_allocated    = 0 -> (known after apply)
                    }
        
                  + nat_gateway_profile {
                      + effective_outbound_ips    = (known after apply)
                      + idle_timeout_in_minutes   = (known after apply)
                      + managed_outbound_ip_count = (known after apply)
                    }
                }
        
              ~ windows_profile {
                  + admin_password = (sensitive value)
                  ~ admin_username = "azureuser" -> (known after apply)
                  + license        = (known after apply)
        
                  + gmsa {
                      + dns_server  = (known after apply)
                      + root_domain = (known after apply)
                    }
                }
            }
        
        Plan: 1 to add, 0 to change, 1 to destroy.
```